{
	"schema_version": "1.6.0",
	"id": "GHSA-hr4f-6jh8-f2vq",
	"modified": "2024-08-21T14:56:46.079008Z",
	"published": "2023-10-18T18:25:58Z",
	"aliases": [
		"CVE-2023-45810",
		"GO-2023-2121"
	],
	"related": [
		"CVE-2023-45810"
	],
	"summary": "OpenFGA DoS vulnerability",
	"details": "## Overview\nOpenFGA is vulnerable to a DoS attack. When a number of ListObjects calls are executed, in some scenarios, those calls are not releasing resources even after a response has been sent, and the service as a whole becomes unresponsive.\n\n## Fix\nUpgrade to v1.3.4. This upgrade is backwards compatible.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Go",
				"name": "github.com/openfga/openfga",
				"purl": "pkg:golang/github.com/openfga/openfga"
			},
			"ranges": [
				{
					"type": "SEMVER",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "1.3.4"
						}
					]
				}
			],
			"database_specific": {
				"source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/10/GHSA-hr4f-6jh8-f2vq/GHSA-hr4f-6jh8-f2vq.json"
			}
		}
	],
	"references": [
		{
			"type": "WEB",
			"url": "https://github.com/openfga/openfga/security/advisories/GHSA-hr4f-6jh8-f2vq"
		},
		{
			"type": "ADVISORY",
			"url": "https://nvd.nist.gov/vuln/detail/CVE-2023-45810"
		},
		{
			"type": "PACKAGE",
			"url": "https://github.com/openfga/openfga"
		},
		{
			"type": "WEB",
			"url": "https://github.com/openfga/openfga/releases/tag/v1.3.4"
		}
	],
	"database_specific": {
		"cwe_ids": [
			"CWE-400"
		],
		"github_reviewed": true,
		"github_reviewed_at": "2023-10-18T18:25:58Z",
		"nvd_published_at": "2023-10-17T23:15:12Z",
		"severity": "HIGH"
	}
}
